File: src/types/appointment.ts
export interface WfAppointment {
  id: number
  hn: string
  apptDate: string
  apptType?: 'inr_check' | 'clinic_visit' | 'urgent'
  status: 'scheduled' | 'completed' | 'missed' | 'cancelled'
  notes?: string
  createdAt: string
  /**
   * Backend-computed: true only when the appointment is in the past, the
   * clinic ran that day, and the patient has no visit record for that day.
   * Use this to decide whether to show "เกินนัด" badges. Optional because
   * older callers that don't use the new query leave it undefined.
   */
  isOverdue?: boolean
}

export interface AppointmentInput {
  hn: string
  apptDate: string
  apptType?: string
  notes?: string
}

export interface AppointmentDayLoad {
  apptDate: string
  scheduledCount: number
}

File: crates/warfarin-core/src/models/appointment.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WfAppointment {
  pub id: i64,
  pub hn: String,
  pub appt_date: String,
  pub appt_type: Option<String>,
  pub status: String,
  pub notes: Option<String>,
  pub created_at: String,
  /// `true` only when the appointment is in the past, the clinic ran on that
  /// day, AND the patient has no `wf_visits` row for that day. Populated by
  /// `get_pending_appointments`; older callers leave it `None` so the field
  /// stays backward-compatible over Tauri IPC.
  #[serde(skip_serializing_if = "Option::is_none", default)]
  pub is_overdue: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AppointmentInput {
  pub hn: String,
  pub appt_date: String,
  pub appt_type: Option<String>,
  pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AppointmentDayLoad {
  pub appt_date: String,
  pub scheduled_count: i64,
}
